### PR TITLE
SLING-10405 update FileVault to 3.5.0

### PR DESCRIPTION
**Pull request by kwin**
SLING-10417 always install packages in strict mode
Get rid of subpackage handling, as dealt with transparently during
package extraction

### CODE REVIEW

File: src/main/java/org/apache/sling/feature/extension/content/ContentHandler.java [hunk 1]
```diff
@@ -90,20 +92,6 @@ private static ExecutionPlanBuilder buildExecutionPlan(Collection<Artifact> arti
 
     }
 
-    private static void extractSubPackages(FSPackageRegistry registry, ExecutionPlanBuilder builder, PackageId pid)
-            throws IOException {
-        Map<PackageId, SubPackageHandling.Option> subPkgs = registry.getInstallState(pid).getSubPackages();
```
Comment (kwin) [inline src/main/java/org/apache/sling/feature/extension/content/ContentHandler.java:95]: This is no longer possible as the getInstallState is a private method. IMHO this shouldn't be necessary at all, as during extraction all subpackages are automatically extracted. No need to flatten those before...
@DominikSuess What was the reason why you extracted the sub packages eagerly?

**Comment by kwin** [comment]
@karlpauls Not all shaded classes are relocated. Is there any rule of thumbs for feature model extensions what should be relocated and also which dependencies you can rely on without embedding it?

**Comment by karlpauls** [comment]
@kwin, I guess the idea is that extensions end-up on the class path for the launcher - hence, it is best if they are self-contained with the exception of whatever the launcher jar provides. As we can't say what extensions will be on the class path, preferably everything that they contain in addition to their own namespace should be shaded.

**Comment by kwin** [comment]
@karlpauls Thanks, should be done in https://github.com/apache/sling-org-apache-sling-feature-extension-content/pull/16/commits/9bd7a12d903e10b14f465c156b1f3be5ff67bce0. I also changed the package name of the relocated packages to make it less likely that the names clash.

**Review by karlpauls — COMMENTED** [review]
Looks good to me but I have the same question as you, @kwin - namely, why was it done in this way to being with wrt. sub packages. It looks like the extract was done for https://issues.apache.org/jira/browse/SLING-8215 - do you think that is addressed by recent vault changes? 
I'm confused as it sounds like https://issues.apache.org/jira/browse/JCRVLT-524 is still open...

**Comment by kwin** [comment]
AFAIK the FSPackageRegistry always correctly dealt with nested subpackages due to the recursion in https://github.com/apache/jackrabbit-filevault/blob/14615ce5647252005f2271b5f5f0351eb91aa78e/vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/FSPackageRegistry.java#L441.

JCRVLT-524 is only about `JcrPackageRegistry` which behaves differently (but this doesn't matter in this context).

**Comment by kwin** [comment]
What about dependency `javax.servlet:javax.servlet-api` (transitive provided dependency of `org.apache.jackrabbit:jackrabbit-spi2dav`). Should that be shaded + relocated as well?

**Comment by karlpauls** [comment]
Is that really needed?

**Comment by kwin** [comment]
I think with https://github.com/apache/sling-org-apache-sling-feature-extension-content/pull/16/commits/c4bf1ced1cb5c6353582422f6d9f566d7443ef68 the shading should be complete (but the JAR is huge). Very hard to tell which dependencies are used at runtime. I wrote an enforcer rule in https://github.com/apache/maven-enforcer/pull/97 to check the class path and the check only emitted the following dependencies missing from the runtime classpath now:

1. org.apache.sling:org.apache.sling.feature.launcher
1. org.osgi:*
1. org.apache.geronimo.specs:geronimo-json_1.1_spec
1. org.apache.felix:org.apache.felix.cm.json
1. org.slf4j:slf4j-api:1.7.25
1. org.apache.felix:org.apache.felix.utils
1. org.apache.felix:org.apache.felix.converter
1. org.apache.sling:org.apache.sling.feature
1. javax.servlet:javax.servlet-api
1. javax.servlet:servlet-api
1. org.jetbrains:annotations

All those are either not necessary (the last three) or provided by the Launcher.

**Review by karlpauls — APPROVED** [review]
LGTM - would be nicer if we could get down the size by only embedding what is needed but I'm not sure I know of a way to figure that out. I guess the is no conditional package with a relocation option?

**Pull request merged**